processor/src/stack/overflow.rs

```diff
@@ -0,0 +1,106 @@
+use super::Felt;
+use std::collections::Bound::{Included, Unbounded};
+use winter_utils::collections::BTreeMap;
+
+// OVERFLOW
+// ================================================================================================
+
+/// Overflow table stores a list of overflow values from the stack.
+/// We create a separate structure to be able to store historical changes
+/// in the overflow table, so we can be able to reconstruct what's in the overflow table
+/// at any given point in time.
+#[derive(Clone)]
+pub struct OverflowTable {
+    active: Vec<Felt>,
+    trace: BTreeMap<usize, Vec<Felt>>,
+    trace_enabled: bool,
+    mutated: bool,
+    clk: usize,
+}
+
+impl OverflowTable {
+    pub fn new(trace_enabled: bool) -> Self {
+        Self {
+            active: Vec::new(),
+            trace: BTreeMap::new(),
+            mutated: false,
+            clk: 0,
+            trace_enabled,
+        }
+    }
+
+    pub fn push(&mut self, value: Felt) {
+        self.mutated = true;
+        self.active.push(value);
+    }
+
+    pub fn pop(&mut self) -> Option<Felt> {
+        self.mutated = true;
+        self.active.pop()
+    }
+
+    pub fn advance_clock(&mut self, clk: usize) {
+        if self.trace_enabled && self.mutated {
+            self.trace.insert(self.clk, self.active.clone());
+            self.mutated = false;
+        }
+        self.clk = clk;
+    }
```

Comment:
Could we simplify this and get rid of `advance_clock()` and `self.mutated` members and do all required updates in `push()` and `pop()`? Or is something preventing this approach?